### PR TITLE
Datetime support by extending Date type (proposal 1)

### LINKED ISSUE #390: Datetime field type
### What would you like to be added?

Since Obsidian Projects already supports date field types, I'd like to suggest the addition of a datetime field type, which keeps compatibility with the former.

More specifically, during the type inference of a field, the common denominator type for a field with either dates (`YYYY-MM-DD`) or datetime values (`YYYY-MM-DDTHH:MM:SS` or `YYYY-MM-DD HH:MM:SS`) should be 'datetime', as opposed to 'text'.

Since Obsidian Projects allows setting date values from certain views (e.g. Table View) using a native HTML `date` input type, that could be changed to `datetime-local` to allow setting time values alongside a date:

HTML edit showing the 'datetime-local' input type

Obsidian Dataview (and probably also the future Datacore) supports datetime values, meaning that using dataview queries as data sources would still play well with the new type.

### Why is this needed?

Having more granular timestamps for projects would help with organizing multiple tasks/projects during a single day. For example, the "Demo project" bundled with the plugin features entries with a "due" field, which could be even better if showcased using datetime values.

This is one thing I'd really like to see in the plugin, as I'm considering switching some workflows to Obsidian from another app, where this kind of feature proved very helpful. 

Thanks for your work on Obsidian Projects!

### LINKED ISSUE #750: Date & Time Property Calendar View
### What would you like to be added?

The current calendar view only handles the Date metadata type. Can this be modified to either:
1. Sort daily by a Time metadata type so that items appear in time blocks on the calendar
2. Use a Date & time type within the calendar view to sort items by day and time.
Dragging items within a day should update their times similar to how dragging items from day to day updates their date.

### Why is this needed?

There is currently no way to sort items on the calendar if there are multiple items in a day. This would allow an Outlook calendar like functionality

### PR DESCRIPTION
**Pull request by Acylation**
Closes #390 and closes #750, a variation of #888

The UI part shares the same solution. A good thing here, by recognizing datetime value as Date type, is that no extra codes are needed for the filtering/coloring.

### CODE REVIEW

File: src/settings/base/settings.ts [hunk 1]
```diff
@@ -146,7 +146,11 @@ export type StringFieldConfig = {
   richText?: boolean;
 };
 
-export type FieldConfig = StringFieldConfig;
+export type DateFieldConfig = {
+  time?: boolean;
+};
+
+export type FieldConfig = StringFieldConfig & DateFieldConfig;
```
Comment (Acylation) [inline src/settings/base/settings.ts:153]: Is this a proper way to combine different type of field configs?

**Comment by GitMurf** [comment]
@Acylation it seems that Dates are behaving differently as in the past, specifically with empty/null dates. In the past I could use a date selector and choose "clear" to not include a date, and then it would work in Projects. But now it seems it will enter "Invalid Date" in the YAML frontmatter which then breaks Obsidian projects and I have to manually open the source file and clear the value out for that YAML date property.

I believe the solution is to update the date value in the YAML frontmatter to just a blank/empty string instead of "Invalid Date".

**Comment by Acylation** [comment]
Got it, will check this case.

**Comment by GitMurf** [comment]
sounds good. let me know if you have any questions or need me to test anything!

**Pull request merged**